source/how-tos/app-development/interactive/conn-params.rst

```diff
@@ -0,0 +1,108 @@
+.. _app-development-interactive-conn-params:
+
+Connection Parameters (connections.yml)
+=======================================
+
+The `connections.yml` configuration file facilitates communication between 
+the backend server and an app, via the ``before.sh.erb``, ``submit.sh.erb``, 
+and ``view.html.erb`` files.
+
+This configuration can be crucial when specific data, such as a ``csrf_token``, 
+is required for operations like logging into RStudio.
```

Comment:
I believe it's `connection.yml` singular.

I'm trying to figure out a way to tighten this phrasing. One thing is it's a 1 way communication from `before.sh.erb -> view.html.erb` so the language should reflect that.

The other thing I think we need to stress is that these values are generated during runtime.

I think the big thing we need to communicate here in the first paragraph is _why_ anyone would need these.

Here's my first stab at what that may look like:
```
App developers can use `conn_params` in `submit.yml.erb` to pass runtime data generated
in the `before.sh.erb` back to the `view.html.erb`. 

This is helpful for 
* Data that is only known _after_ the job submits and starts running.
* Data that needs to be used to connect to the application.
```

Reply:
> Well what confused me was that I saw things like password and host in the Rstudio app being used in the view that hat not been exported.

Now that you mention it, I suspect it's because we `source` the `before.sh.erb`. I think when you `source` things you don't have to `export`?

I see them being defined, just not exported.

In any case - I think my comment is more around making sure we explain the use case before details of implementation.